src/main/java/edu/cornell/kfs/fp/document/validation/impl/CuDisbursementVoucherAccountingLineTotalsValidation.java

```diff
@@ -65,7 +76,65 @@ public boolean validate(AttributedDocumentEvent event) {
             }
         }
 
+        if (dvDocument instanceof RecurringDisbursementVoucherDocument) {
+            RecurringDisbursementVoucherDocument recurringDV = (RecurringDisbursementVoucherDocument) dvDocument;
+            if (!doesAccountingLineTotalEqualDVTotalDollarAmount(recurringDV)) {
+                String propertyName = KFSPropertyConstants.DOCUMENT + "." + KFSPropertyConstants.DISB_VCHR_CHECK_TOTAL_AMOUNT;
+                GlobalVariables.getMessageMap().putError(propertyName, CUKFSKeyConstants.ERROR_DV_CHECK_TOTAL_MUST_EQUAL_ACCOUNTING_LINE_TOTAL);
+                return false;
+            }
+            if(StringUtils.isEmpty(recurringDV.getDisbVchrCheckStubText()) && !recurringDV.getSourceAccountingLines().isEmpty()) {
+                String propertyName = KFSPropertyConstants.DOCUMENT + "." + KFSPropertyConstants.DISB_VCHR_CHECK_STUB_TEXT;
+                GlobalVariables.getMessageMap().putError(propertyName, CUKFSKeyConstants.ERROR_DV_CHECK_STUB_REQUIRED);
+                return false;
+            }
+            if(!checkAccountingLineEndDateValid(recurringDV)) {
+                return false;
+            }
+        }
+
        return super.validate(event);
     }
+    
+    private boolean doesAccountingLineTotalEqualDVTotalDollarAmount(RecurringDisbursementVoucherDocument recurringDV) {
+        KualiDecimal calculatedTotal = KualiDecimal.ZERO;
+        for (Object accountingLine : recurringDV.getSourceAccountingLines()) {
+            ScheduledSourceAccountingLine scheduledAccountingLine = (ScheduledSourceAccountingLine)accountingLine;	
+            calculatedTotal = calculatedTotal.add(scheduledAccountingLine.getAmount());
+        }
+        return  calculatedTotal.equals(recurringDV.getTotalDollarAmount());
+    }
+
+    private boolean checkAccountingLineEndDateValid(RecurringDisbursementVoucherDocument recurringDV) {
+        boolean valid = true;
+        int counter = 0;
+        Date maximumScheduledAccountingLineEndDate = getScheduledAccountingLineService().getMaximumScheduledAccountingLineEndDate();
+        SimpleDateFormat dateFormater = new SimpleDateFormat("MM/dd/yyyy");
```

Comment:
Seems like maybe there should be constants for some of these string literals (I see three in this method).